packages/devtools/lib/src/framework/framework.dart

```diff
@@ -499,10 +499,12 @@ class ConnectDialog {
     );
 
     if (connected) {
-      // Re-write the url to include the new port
+      // Re-write the url to include the new port. Keep existing query params.
       final Location location = window.location;
       Uri uri = Uri.parse(location.href);
-      uri = uri.replace(queryParameters: {'port': port.toString()});
+      uri = uri.replace(
+          queryParameters: {'port': port.toString()}
```

Comment:
tiny nit: should we add ['port'] = port.toString()
instead so the port shows as the last query parameter which might be a little nicer.

Reply:
for some reason doing `uri.queryParameters..addAll({'port': port.toString()})` doesn't work.